torch/_inductor/graph.py

```diff
@@ -625,7 +627,10 @@ def register_buffer(self, buffer: ir.Buffer):
         self.buffers.append(buffer)
         self.name_to_buffer[name] = buffer
         # Skip empty CPU tensor so that CUDA graphs can succeed, see https://github.com/pytorch/pytorch/pull/114144
-        if not isinstance(buffer, ir.ComputedBuffer) or not buffer.is_zero_elements():
+        if (
+            not (isinstance(buffer, ir.ComputedBuffer) and buffer.is_zero_elements())
+            and buffer.get_device() is not None
```

Comment:
Mostly a curiosity question, but what's the change here for? (I think I would have naively thought that effect tokens wouldn't change how inductor handles buffers, since the new `prims._new_token` op should just create a temporary, but idk)

Reply:
Ah the reason is because calling `print` has no return, so the buffer's device is `None` and `add_device_info` doesn't handle the case where the device is `None`. So I added some extra casing here.

Reply:
ah I see - so EffectToken op outputs show up as buffers in inductor IR (and they are effectively None / never used). Thanks!